docs/docs/creating-and-modifying-pages.md

```diff
@@ -0,0 +1,71 @@
+---
+title: "Creating and modifying pages"
+---
+
+Gatsby makes it easy to programatically control your pages.
+
+Pages can be created in three ways:
+
+* In your site code by implementing the API `createPages`
+* Gatsby core automatically turns React compnents in `src/pages` into pages
+* Plugins can also implement `createPages` and create pages for you
+
+## Modifying pages created by core or plugins
+
+Gatsby core and plugins can automatically create pages for you. Sometimes
+the default isn't quite what you want and you need to modify the created
+page objects.
+
+### Removing trailing slashes
+A common reason for needing to modify automatically created pages is to remove
+trailing slashes.
+
+To do this, in your site's `gatsby-node.js` add code
+similar to the following:
+
+```javascript
+// Implement the Gatsby API “onUpsertPage”. This is
+// called after every page is created.
+exports.onUpsertPage = async ({ page, boundActionCreators }) => {
+  const { upsertPage, deletePageByPath } = boundActionCreators
+
+  // Remove trailing slash
+  const oldPath = page.path
+  page.path = page.path.replace(/\/$/, "")
```

Comment:
Oh right, forgot about that too... yeah probably we just need to make Gatsby not call the plugin that called `upsertPage` the first time... I'll add this in a sec.